merge_source_location/P1208.bs

```diff
@@ -91,45 +91,44 @@ Date: 2018-09-21
+Note: A proposal [[P0555R0]] in 2017 to make use of `string_view` in `source_location` has been withdrawn by their authors once `string_view`
+gained a constexpr constructor, allowing the use of `source_location` in constexpr contexts.
 
 # A few design changes
 
-During an early review of this papers, some interests for a few design changes arose
+During an early review of this papers, some interest in a few design changes arose:
 
-## Enforcing a size for `source_location` objects ?
+## Enforcing a size for `source_location` objects?
 
-`source_location` as currently implemented by Clang And GCC is a structure encapsulating all its members (file, function, line, column),
-and so its size is roughly 3 * sizeof(void*).
+`source_location` as currently implemented by Clang and GCC is a structure encapsulating all its members (file, function, line, column),
+and so its size is roughly `3 * sizeof(void*)`.
 
-A note state
-> [ Note: The intent of source_location is to have a small size and efficient copying. — end note ]
+A note states:
+> [ Note: The intent of `source_location` is to have a small size and efficient copying. — end note ]
 
-However, there seems to be some interest for having `source_location` be guaranteed `sizeof(void*)`.
-Notably, reviewers wonder if `source_location` should be embedded in a yet to be proposed `std::error`.
+However, there seems to be some interest in having `source_location` be guaranteed `sizeof(void*)`.
+
+Some reviewers wonder if `source_location` should be embedded in a yet-to-be-proposed `std::error`.
 Herb Sutter and Niall Douglas pointed out that `std::error` is meant to handle recoverable errors
 rather than to communicate an error to developers (see Herb Sutter's talk on error handling CppCon 2018).
 And while we agree strongly with this sentiment, it's worth noting that a subset of the C committee
-is interested in having source information in whatever deterministic exception handling they are working towards;
-
-While, as pointed out by Niall, it's unlikely the two languages will have directly interoperable mechanisms,
+is interested in having source information in whatever deterministic exception handling they are working towards.
+As pointed out by Niall, it's unlikely that C and C++ will have directly interoperable mechanisms; yet,
 having a guaranteed size for `source_location` might make such interoperability easier should a need for it arise.
 
-Baring that, `source_location` being an error reporting tool targeted at human developers,
+Barring that, `source_location` is an error reporting tool targeted at human developers,
 and the cost of any error reporting system is many orders of magnitude more resource-demanding than copying
-3 pointers around, we do not think there is a strong incentive to guaranteeing the size of that type
+3 pointers around. We do not think there is a strong incentive to guaranteeing the size of that type
 or restricting the way it is implemented.
```

Comment:
This seems like an *extremely* weak argument against a 3x cost savings for `source_location`.

Sure, putting `source_location` inside an error type is a bad idea. But surely we do expect programmers to pass around objects of type `source_location` *somehow*, right? Otherwise you wouldn't care about the type's value semantics, or even care about the type existing at all. So, given that we're going to be storing and passing around these objects... wouldn't it be nice if they were 3x smaller, and fit in a single machine register? What's the argument *against* efficiency here?

Reply:
Philosophically, I'm okay leaving it as QoI. Pragmatically, I worry that vendors are already out there implementing it as 3x larger than it "should" be. As long as `source_location` is experimental, those vendors should be happy to take patches improving the performance 3x, even if it is an ABI break. Once `source_location` is in the Standard, I think the cost of an ABI break goes way up.